File: Day267/minCostCLimbingStairs.java
public class minCostCLimbingStairs {
    public static int minCostClimbingStairs(int cost[]) {
        int answer[] = new int[cost.length+1];
        answer[cost.length-1] = cost[cost.length-1];
        answer[cost.length-2] = cost[cost.length-2]; 
        for (int i=cost.length-3; i>=0; i--) {
            answer[i] = cost[i] + Math.min(answer[i+1], answer[i+2]);
        }
        // System.out.println(Arrays.toString(answer));
        return Math.min(answer[0], answer[1]);
    }
    public static void main(String[] args) {
        int arr1[] = {10, 15, 20};
        int arr2[] = {1,100,1,1,1,100,1,1,100,1};
        System.out.println(minCostClimbingStairs(arr1));
        System.out.println(minCostClimbingStairs(arr2)); 
    }
}

File: Day267/minCostCLimbingStairs.go
package main

import (
	"fmt"
	"math"
)

func minCostClimbingStairs(arr []int) int {
	arr = append(arr, 0)
	for i:= len(arr) - 3; i >= 0; i-- {
		arr[i] += int(math.Min(float64(arr[i+1]), float64(arr[i+2])))
	}
	return int(math.Min(float64(arr[0]), float64(arr[1])))
}

func main() {
	arr1 := []int{10, 15, 20}
	arr2 := []int{1,100,1,1,1,100,1,1,100,1}
	fmt.Println(minCostClimbingStairs(arr1))
	fmt.Println(minCostClimbingStairs(arr2))
}